tests/integration/functions/types_test.ts

```diff
@@ -0,0 +1,199 @@
+import { assert, IsExact } from "../../../src/dev_deps.ts";
+import {
+  EnrichedSlackFunctionHandler,
+  FunctionRuntimeType,
+} from "../../../src/functions/types.ts";
+import { DefineFunction, DefineProperty, Schema } from "../../../src/mod.ts";
+import { SlackFunctionTester } from "../../../src/functions/tester/mod.ts";
+
+Deno.test("FunctionRuntimeType should abe able to provide a usable type of a DefineFunction return object", () => {
+  const TestFn = DefineFunction({
+    callback_id: "test_function",
+    title: "Test function",
+    source_file: "functions/test_function.ts",
+    input_parameters: {
+      properties: {
+        bool: {
+          type: Schema.types.boolean,
+        },
+        int: {
+          type: Schema.types.integer,
+        },
+        num: {
+          type: Schema.types.number,
+        },
+        string: {
+          type: Schema.types.string,
+        },
+        arr: {
+          type: Schema.types.array,
+          items: {
+            type: Schema.types.boolean,
+          },
+        },
+        obj: DefineProperty({
+          type: Schema.types.object,
+          properties: {
+            bool: {
+              type: Schema.types.boolean,
+            },
+          },
+          required: ["bool"],
+        }),
+      },
+      required: ["bool", "int", "num", "string", "arr", "obj"],
+    },
+    output_parameters: {
+      properties: {
+        bool: {
+          type: Schema.types.boolean,
+        },
+        int: {
+          type: Schema.types.integer,
+        },
+        num: {
+          type: Schema.types.number,
+        },
+        string: {
+          type: Schema.types.string,
+        },
+        arr: {
+          type: Schema.types.array,
+          items: {
+            type: Schema.types.boolean,
+          },
+        },
+        obj: DefineProperty({
+          type: Schema.types.object,
+          properties: {
+            bool: {
+              type: Schema.types.boolean,
+            },
+          },
+          required: ["bool"],
+        }),
+      },
+      required: ["bool", "int", "num", "string", "arr", "obj"],
+    },
+  });
+
+  type Actual = FunctionRuntimeType<typeof TestFn>;
+
+  const expectedParams = {
+    bool: true,
+    int: 0,
+    num: 0.1,
+    string: "",
+    arr: [true],
+    obj: {
+      bool: true,
+    },
+  };
+
+  const validHandler: EnrichedSlackFunctionHandler<typeof TestFn.definition> =
+    () => {
```

Comment:
One thing you might want to assert on here is that the type of the arguments provided to the function handler match the `Actual.args` type.

```suggestion
    (args) => {
      assert<IsExact<typeof args, Actual["args"]>>(true);
```

I think this is a slightly better testing approach than testing the type of what is returned by `createContext` below, because `createContext` is a testing utility we wrote, that might have some shortcuts or use other code paths that are not user/developer-facing. Therefore, asserting on the output of `createContext` tests our own test utilities rather than testing the parts that app devs interface with.

Reply:
Good catch based on this I think I've been able to better implement the assertions of this file